sos/report/__init__.py

```diff
@@ -373,6 +378,34 @@ def _get_hardware_devices(self):
         }
         # TODO: enumerate network devices, preferably with devtype info
 
+    def _check_container_runtime(self):
+        """Check the loaded container runtimes, and the policy default runtime
+        (if set), against any requested --container-runtime value. This can be
+        useful for systems that have multiple runtimes, such as RHCOS, but do
+        not have a clearly defined 'default' (or one that is determined based
+        entirely on configuration).
+        """
+        if self.opts.container_runtime != 'auto':
+            crun = self.opts.container_runtime.lower()
+            if crun in ['none', 'off', 'diabled']:
+                self.policy.runtimes = {}
+                self.soslog.info(
+                    "Disabled all container runtimes per user option."
+                )
+            elif not self.policy.runtimes:
+                msg = ("WARNING: No container runtimes are active, ignoring "
+                       "option to set default runtime to '%s'\n" % crun)
+                self.soslog.warn(msg)
+            elif crun not in self.policy.runtimes.keys():
+                raise Exception("Cannot use container runtime '%s': no such "
+                                "runtime detected." % crun)
```

Comment:
Maybe extend the message by `self.policy.runtimes.keys()` ? It might be valuable to know what runtimes are present (or to be more precise: what `sos` detected as present), to understand why user chosen non-existing runtime?